### PR TITLE
KAFKA-5548 - Extended validation for SchemaBuilder methods.

### PR DESCRIPTION
**Pull request by jcustenborder**
More input validation for SchemaBuilder methods.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5857/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5843/
Test PASSed (JDK 8 and Scala 2.12).

File: connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java [hunk 1]
```diff
@@ -319,6 +319,10 @@ public static SchemaBuilder struct() {
     public SchemaBuilder field(String fieldName, Schema fieldSchema) {
         if (type != Type.STRUCT)
             throw new SchemaBuilderException("Cannot create fields on type " + type);
+        if (null == fieldName || fieldName.isEmpty())
+            throw new SchemaBuilderException("fieldName cannot be null.");
```
Comment (ewencp) [inline connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java:323]: Possibly my favorite aspect of this PR is that you have reminded me that `SchemaBuilderException` is even a thing that exists...

**Review by ewencp — APPROVED** [review]
LGTM, probably a great supportability improvement. thanks!

**Pull request closed without merging**